### PR TITLE
Add push notifications

### PR DESCRIPTION
**Pull request by upacyxou**
- add sending real notifications

### CODE REVIEW

File: src/helpers/apn.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+const pathToKey = resolve(cwd(), 'APN.p8')
+
+export default new Provider({
+  token: {
+    key: pathToKey,
+    keyId: env.APN_KEY_ID,
+    teamId: env.APN_TEAM_ID,
+  },
+  production: env.isProduction,
```
Comment (T-Damer) [inline src/helpers/apn.ts:14]: How do we set this? 
Should we use `env.NODE_ENV === 'production'`
?

File: src/helpers/firebase.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import * as admin from 'firebase-admin'
+import { cwd } from 'process'
+import { initializeApp } from 'firebase-admin/app'
+import { resolve } from 'path'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const serviceAccount = require(resolve(cwd(), 'firebase-account.json'))
+
+const app = initializeApp({
```
Comment (T-Damer) [inline src/helpers/firebase.ts:9]: Why don't we simply `export default` this without `const app`?

File: src/helpers/getObssContract.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { OBSSStorage__factory } from '@big-whale-labs/obss-storage-contract'
+import defaultProvider from '@/helpers/defaultProvider'
+import env from '@/helpers/env'
+
+export default function () {
```
Comment (T-Damer) [inline src/helpers/getObssContract.ts:5]: What if we'll:
```TS
export default OBSSStorage__factory.connect(
    env.OBSS_STORAGE_CONTRACT,
    defaultProvider
  )
```

Comment (T-Damer) [reply src/helpers/getObssContract.ts:5]: Not sure if it will work as expected

File: src/helpers/runNotificationListener.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { TokenModel } from '@/models/Token'
+import getObssContract from '@/helpers/getObssContract'
+import sendAppleNotification from '@/helpers/sendAppleNotification'
+import sendGoogleNotification from '@/helpers/sendGoogleNotification'
+
+const apnRegex = /^[a-f0-9]{64}$/
```
Comment (T-Damer) [inline src/helpers/runNotificationListener.ts:6]: Btw, if we already have a regex for this, should we use it in `env.ts`?
image

File: src/helpers/apn.ts (hunk 1 above)
Comment (upacyxou) [reply src/helpers/apn.ts:14]: no, it's literallt same

File: src/helpers/runNotificationListener.ts (hunk 4 above)
Comment (upacyxou) [reply src/helpers/runNotificationListener.ts:6]: no, not related

**Review by T-Damer — APPROVED** [review]

**Pull request merged**